netbox/extras/querysets.py

```diff
@@ -151,3 +153,14 @@ def _get_config_context_filters(self):
         )
 
         return base_query
+
+
+class ObjectChangeQuerySet(RestrictedQuerySet):
+
+    def valid_models(self):
+        # Exclude any change records which refer to an instance of a model that's no longer installed. This
+        # can happen when a plugin is removed but its data remains in the database, for example.
+        content_type_ids = set(
+            ct.pk for ct in ContentType.objects.get_for_models(*apps.get_models()).values()
```

Comment:
I'm still getting hung up on this approach. While it does work, it seems far from ideal, and I worry about the performance impact.

Reply:
How about using `Subquery` here? I'm not exactly sure if that's going to improve anything performance wise

Reply:
I played around in django shell with subquery and you can try something like this

```
ObjectChange.objects.filter(changed_object_type_id__in=Subquery(ContentType.objects.filter(model__in=[m._meta.model_name for m in apps.get_models()]).values_list("pk")))
```

Reply:
It's the base query that concerns me, filtering for `id IN [1, 2, 3, ...]`. It just seems really inefficient, but I haven't been able to come up with a more clever solution.